File: shared/src/main/java/ru/privatenull/pnauth/api/AuthDialogApi.java
package ru.privatenull.pnauth.api;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

/** Client UI preference and capability negotiation. */
public interface AuthDialogApi {
    DialogPreference dialogPreference(UUID uniqueId);
    CompletableFuture<AuthResult> setDialogPreference(UUID uniqueId, DialogPreference preference);
    boolean shouldUseDialog(UUID uniqueId, int clientProtocol, boolean platformSupportsDialogs);

    default boolean shouldUseCommandFallback(UUID uniqueId, int clientProtocol, boolean platformSupportsDialogs) {
        return !shouldUseDialog(uniqueId, clientProtocol, platformSupportsDialogs);
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthDialogApi.kt
package ru.privatenull.pnauth.api

import java.util.UUID
import java.util.concurrent.CompletableFuture

/** Client UI preference and capability negotiation. */
interface AuthDialogApi {
    fun dialogPreference(uniqueId: UUID): DialogPreference
    fun setDialogPreference(uniqueId: UUID, preference: DialogPreference): CompletableFuture<AuthResult>
    fun shouldUseDialog(uniqueId: UUID, clientProtocol: Int, platformSupportsDialogs: Boolean): Boolean

    fun shouldUseCommandFallback(uniqueId: UUID, clientProtocol: Int, platformSupportsDialogs: Boolean): Boolean {
        return !shouldUseDialog(uniqueId, clientProtocol, platformSupportsDialogs)
    }
}
